src/request.js

```diff
@@ -206,22 +206,20 @@ export const getNodeRequestOptions = request => {
 
 	const search = getSearch(parsedURL);
 
-	// Manually spread the URL object instead of spread syntax
-	const requestOptions = {
+	// Pass the full URL directly to request(), but overwrite the following 
+	// options:
+	const options = {
+		// Overwrite search to retain trailing ? (issue #776)
 		path: parsedURL.pathname + search,
-		pathname: parsedURL.pathname,
-		hostname: parsedURL.hostname,
-		protocol: parsedURL.protocol,
-		port: parsedURL.port,
-		hash: parsedURL.hash,
-		search: parsedURL.search,
-		query: parsedURL.query,
-		href: parsedURL.href,
```

Comment:
For those who are curious why these can be removed: 
- Since we're passing the URL to `request()`, we are now using the Node.js standard library URL parser
- The standard library URL parser effectively [expands the URL into the same set of options](https://github.com/nodejs/node/blob/1c4df352b77514e93a46fb27e996755d45a0b530/lib/internal/url.js#L1388).

Reply:
Oh! This led me to a side effect of this PRs' changes: [Basic auth](https://github.com/nodejs/node/blob/1c4df352b77514e93a46fb27e996755d45a0b530/lib/internal/url.js#L1405) (`username:password@hostname`) is now supported implictly by `node-fetch`. 

Should this be a concern?

Seems to be allowed in the [`fetch()` spec, §4.2 step 8.19.2.3.](https://fetch.spec.whatwg.org/#http-network-or-cache-fetch)

Reply:
👍